### PR TITLE
Feature resample histogram

### PR DESCRIPTION
**Pull request by nbiederbeck**
supersedes #64 and adds #49

### CODE REVIEW

**Comment by nbiederbeck** [comment]
I am trying ..

File: pyirf/binning.py [hunk 1]
```diff
@@ -128,3 +129,58 @@ def create_histogram_table(events, bins, key="reco_energy"):
             events[key], bins, weights=events["weight"]
         )
     return hist
+
+
+def resample_histogram(data, old_edges, new_edges, axis=0):
```
Comment (MaxNoe) [inline pyirf/binning.py:134]: maybe add a `1d`?

File: docs/irf/index.rst [hunk 2]
```diff
@@ -15,10 +15,12 @@ the angular cuts.
 Energy Dispersion Matrix
 ------------------------
 
-The migration matrix, ratio of the reconstructed energy over the true energy
+The energy dispersion matrix, ratio of the reconstructed energy over the true energy
 as a function of the true energy, is computed with the events passing the
 threshold of the best cutoff plus the angular cuts.
 
+The corresponding, sparse energy migration matrix can be build from the dispersion matrix.
```
Comment (MaxNoe) [inline docs/irf/index.rst:22]: Sparse is not part of the definion. it's more like an expected property of the migration matrix.

The real difference is in the second axis: dispersion as `(e_reco / e_true)` , migration has `e_reco`

**Review by MaxNoe — CHANGES_REQUESTED** [review]
Please add a test and fix the behaviour for bins that include under/overflow bin.

When I tested this locally with edisp that had under/overflow bins, all migration bins were `np.nan`

**Comment by nbiederbeck** [comment]
> Does this now mean it is required to have an overflow bin?

No. There is a test with and without overflow bins. `np.nan_to_num` does nothing if there are no over/underflow bins

> And shouldn't it be `old_edges[-1]`?

`old_edges[-1] == np.inf` (with overflow bins), so we chose the second last bin which is not inf

**Comment by nbiederbeck** [comment]
Travis still fails while downloading the test data

**Comment by MaxNoe** [comment]
Ah, dam, this is because PRs from forks don't have access to the credentials.

**Comment by HealthyPear** [comment]
> Ah, dam, this is because PRs from forks don't have access to the credentials.

Pardon my ignorance, but why would it matter?
Isn't this password just a string? if one has it it downloads otherwise not
or there is some other linked with information with it?

**Comment by MaxNoe** [comment]
The download needs a password. That password is stored using travis' credential system. 

Build from pull requests don't have access to these credentials, because they could just add something like `echo $PASSWORD` to the .travis.yml to get the password.

So, only builds from the repo itself have access to these credentials for security reasons.

**Comment by nbiederbeck** [comment]
after pushing to the main repo I need to open another PR ?

**Comment by MaxNoe** [comment]
Probably, yes. I don't think there is the possibility to change the branch of a PR
